web-ui/src/components/guides/GuidesPanel.jsx

```diff
@@ -1,40 +1,82 @@
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import PdfIcon from '@mui/icons-material/PictureAsPdf';
-import Card from '@mui/material/Card';
-import CardHeader from '@mui/material/CardHeader';
-import List from '@mui/material/List';
-import './GuidesPanel.css';
+import { Card, CardHeader, List } from '@mui/material';
+
+import { AppContext } from '../../context/AppContext.jsx';
+import { selectCsrfToken, selectRoles } from '../../context/selectors.js';
+import { getDocumentsForRoleId } from '../../api/document.js';
+
 import GuideLink from './GuideLink';
+import './GuidesPanel.css';
 
-const GuidesPanel = () => {
-  const teamMemberPDFs = [
-    {
-      name: 'Expectations Discussion Guide for Team Members'
-    },
-    {
-      name: 'Expectations Worksheet'
-    },
-    {
-      name: 'Feedback Discussion Guide for Team Members'
-    },
-    {
-      name: 'Development Discussion Guide for Team Members'
-    },
-    {
-      name: 'Individual Development Plan'
+const fallbackPdfs = [
+  {
+    id: '1-member',
+    name: 'Expectations Discussion Guide for Team Members',
+    url: '/pdfs/Expectations_Discussion_Guide_for_Team_Members.pdf'
+  },
+  {
+    id: '2-member',
+    name: 'Expectations Worksheet',
+    url: '/pdfs/Expectations_Worksheet.pdf'
+  },
+  {
+    id: '3-member',
+    name: 'Feedback Discussion Guide for Team Members',
+    url: '/pdfs/Feedback_Discussion_Guide_for_Team_Members.pdf'
+  },
+  {
+    id: '4-member',
+    name: 'Development Discussion Guide for Team Members',
+    url: '/pdfs/Development_Discussion_Guide_for_Team_Members.pdf'
+  },
+  {
+    id: '5-member',
+    name: 'Individual Development Plan',
+    url: '/pdfs/Individual_Development_Plan.pdf'
+  }
+];
+
+export const fetchDocumentsForRole = (roleName, allRoles, csrf, setDocuments, fallback, mockuments) => {
+  useEffect(() => {
+    async function getDocuments() {
+      if (mockuments && mockuments.length > 0) {
+        setDocuments(mockuments);
+      } else {
+        const memberRoleId = allRoles.find(role => role.role === roleName)?.id;
+        const res = await getDocumentsForRoleId(memberRoleId, csrf);
+        let responseBody = res.payload && res.payload.data && !res.error
```

Comment:
You can use `const` instead of `let` here. That is preferred to communicate that you don't plan to change the value later.

Reply:
Also, this can be shorted a bit using the "optional chaining" operator.
res.payload && res.payload.data
can be replaced by just
res.payload?.data

Reply:
Cool, done in https://github.com/objectcomputing/check-ins/pull/2522/commits/edd13f14cde173d1e928a002cd7db11e018ad86c